INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:Загрузка главной страницы
INFO:werkzeug:127.0.0.1 - - [03/May/2022 01:10:31] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 01:10:31] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Загрузка главной страницы
INFO:werkzeug:127.0.0.1 - - [03/May/2022 01:10:32] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 01:10:32] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Загрузка главной страницы
INFO:werkzeug:127.0.0.1 - - [03/May/2022 01:10:32] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 01:10:32] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Выполяется поиск
INFO:werkzeug:127.0.0.1 - - [03/May/2022 01:10:41] "GET /search?s=слишком HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 01:10:41] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Выполяется поиск
INFO:werkzeug:127.0.0.1 - - [03/May/2022 01:11:21] "GET /search?s=еда HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 01:11:21] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Выполяется поиск
INFO:werkzeug:127.0.0.1 - - [03/May/2022 21:10:29] "GET /search?s=еда HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 21:10:29] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Выполяется поиск
INFO:werkzeug:127.0.0.1 - - [03/May/2022 21:10:39] "GET /search?s=горы HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 21:10:39] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Выполяется поиск
INFO:werkzeug:127.0.0.1 - - [03/May/2022 21:10:52] "GET /search?s=поля HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 21:10:52] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:Выполяется поиск
INFO:werkzeug:127.0.0.1 - - [03/May/2022 21:13:10] "GET /search?s=поля HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 21:13:10] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Выполяется поиск
INFO:werkzeug:127.0.0.1 - - [03/May/2022 21:13:23] "GET /search?s=поля HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 21:13:23] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Выполяется поиск
INFO:werkzeug:127.0.0.1 - - [03/May/2022 21:13:23] "GET /search?s=поля HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 21:13:23] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Выполяется поиск
INFO:werkzeug:127.0.0.1 - - [03/May/2022 21:13:24] "GET /search?s=поля HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 21:13:24] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:Выполяется поиск
INFO:werkzeug:127.0.0.1 - - [03/May/2022 21:15:21] "GET /search?s=хотя HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 21:15:21] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:Выполяется поиск
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:04:08] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:04:08] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Выполяется поиск
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:04:16] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:04:16] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Выполяется поиск
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:04:16] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:04:16] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Загрузка главной страницы
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:04:33] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:04:33] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:11:55] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:11:55] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:11:55] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:11:55] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:11:56] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:11:56] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:12:09] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:12:09] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:12:25] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:12:25] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:12:26] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:12:26] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:12:26] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:12:26] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:12:26] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:12:26] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:12:26] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:12:26] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:12:57] "GET /search?s=в+бурю HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:12:57] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:16:47] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:16:47] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:16:48] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:16:48] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:16:48] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:16:49] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:16:49] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:16:49] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:16:49] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:16:49] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:16:50] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:16:50] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:28:55] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:28:55] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:29:53] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:29:53] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:29:55] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:29:55] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:30:01] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:30:01] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:30:18] "[33mPOST /post_uploaded.html HTTP/1.1[0m" 404 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:38:05] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:38:05] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:38:07] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:38:07] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:38:11] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:38:11] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:38:12] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:38:12] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:38:14] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:38:14] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:38:20] "[33mPOST /post HTTP/1.1[0m" 404 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:45:15] "GET /search?s=слишком HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:45:15] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:45:21] "[33mGET /post HTTP/1.1[0m" 404 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:45:33] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:45:33] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:45:55] "[33mPOST /post HTTP/1.1[0m" 404 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:45:58] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:45:58] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:46:02] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:46:02] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:46:05] "[33mGET /post HTTP/1.1[0m" 404 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:47:16] "[33mGET /post HTTP/1.1[0m" 404 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:47:19] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:47:19] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:48:00] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:48:00] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:48:04] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:48:04] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:48:08] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:48:08] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
ERROR:main:Exception on /posts [POST]
Traceback (most recent call last):
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:/Users/Admin/PycharmProjects/pythonProject2/main.py", line 36, in created_new_post
    return render_template("post_uploadet.html")
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\templating.py", line 149, in render_template
    ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 1081, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 1010, in get_template
    return self._load_template(name, globals)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 969, in _load_template
    template = self.loader.load(self, name, self.make_globals(globals))
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\loaders.py", line 126, in load
    source, filename, uptodate = self.get_source(environment, name)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\templating.py", line 59, in get_source
    return self._get_source_fast(environment, template)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\templating.py", line 95, in _get_source_fast
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: post_uploadet.html
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:48:25] "[35m[1mPOST /posts HTTP/1.1[0m" 500 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:48:55] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:48:55] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:48:56] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [03/May/2022 23:48:56] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:27:11] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:27:12] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:27:15] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:27:15] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:27:19] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:27:19] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:root:No data uploaded
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:27:28] "POST /posts HTTP/1.1" 200 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:27:31] "GET /posts HTTP/1.1" 200 -
INFO:root:page add post
INFO:root:No data uploaded
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:27:36] "POST /posts HTTP/1.1" 200 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:27:38] "GET /posts HTTP/1.1" 200 -
INFO:root:page add post
INFO:root:No data uploaded
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:27:50] "POST /posts HTTP/1.1" 200 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:27:53] "GET /posts HTTP/1.1" 200 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:28:06] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:28:06] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:root:No data uploaded
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:28:14] "POST /posts HTTP/1.1" 200 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:28:16] "GET /posts HTTP/1.1" 200 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:28:21] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:28:21] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:root:No data uploaded
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:34:19] "POST /posts HTTP/1.1" 200 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:34:21] "GET /posts HTTP/1.1" 200 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:34:24] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:34:24] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:34:40] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:34:40] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:34:41] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:34:41] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:root:No data uploaded
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:34:54] "POST /posts HTTP/1.1" 200 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:34:55] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:36:58] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:36:58] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
ERROR:main:Exception on /posts [POST]
Traceback (most recent call last):
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:/Users/Admin/PycharmProjects/pythonProject2/main.py", line 42, in created_new_post
    return render_template("post_uploadet.html")
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\templating.py", line 149, in render_template
    ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 1081, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 1010, in get_template
    return self._load_template(name, globals)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 969, in _load_template
    template = self.loader.load(self, name, self.make_globals(globals))
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\loaders.py", line 126, in load
    source, filename, uptodate = self.get_source(environment, name)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\templating.py", line 59, in get_source
    return self._get_source_fast(environment, template)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\templating.py", line 95, in _get_source_fast
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: post_uploadet.html
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:37:05] "[35m[1mPOST /posts HTTP/1.1[0m" 500 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:A search is being performed
ERROR:main:Exception on /search [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:/Users/Admin/PycharmProjects/pythonProject2/main.py", line 28, in search_posts
    return render_template("post_list.html", posts=functions.search_post(s), s=s)
TypeError: search_post() missing 1 required positional argument: 'path'
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:46:21] "[35m[1mGET /search?s=кот HTTP/1.1[0m" 500 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:48:00] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 00:48:00] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [04/May/2022 22:23:26] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 22:23:26] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [04/May/2022 22:23:28] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 22:23:29] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
ERROR:main:Exception on /posts [POST]
Traceback (most recent call last):
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:/Users/Admin/PycharmProjects/pythonProject2/main.py", line 48, in created_new_post
    return render_template("post_uploadet.html")
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\templating.py", line 149, in render_template
    ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 1081, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 1010, in get_template
    return self._load_template(name, globals)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 969, in _load_template
    template = self.loader.load(self, name, self.make_globals(globals))
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\loaders.py", line 126, in load
    source, filename, uptodate = self.get_source(environment, name)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\templating.py", line 59, in get_source
    return self._get_source_fast(environment, template)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\templating.py", line 95, in _get_source_fast
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: post_uploadet.html
INFO:werkzeug:127.0.0.1 - - [04/May/2022 22:23:36] "[35m[1mPOST /posts HTTP/1.1[0m" 500 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [04/May/2022 22:35:24] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 22:35:24] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
ERROR:main:Exception on /search [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:/Users/Admin/PycharmProjects/pythonProject2/main.py", line 28, in search_posts
    return render_template("post_list.html", posts=functions.search_post(s, data_file), s=s)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\functions.py", line 16, in search_post
    list_post = load_json_fliles(path)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\functions.py", line 6, in load_json_fliles
    return json.load(file)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python38\lib\json\__init__.py", line 293, in load
    return loads(fp.read(),
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python38\lib\encodings\cp1251.py", line 23, in decode
    return codecs.charmap_decode(input,self.errors,decoding_table)[0]
UnicodeDecodeError: 'charmap' codec can't decode byte 0x98 in position 875: character maps to <undefined>
INFO:werkzeug:127.0.0.1 - - [04/May/2022 22:35:27] "[35m[1mGET /search?s=кот HTTP/1.1[0m" 500 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [04/May/2022 22:36:52] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 22:36:52] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:03:52] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:03:52] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:03:55] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:03:55] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:04:09] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:04:09] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:04:10] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:04:10] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:04:10] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:04:10] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:04:10] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:04:10] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:04:11] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:04:11] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:04:31] "GET /search?s=cat HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:04:32] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:04:54] "GET /search?s=слишком HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:04:54] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
werkzeug - INFO -  * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
werkzeug - INFO - 127.0.0.1 - - [04/May/2022 23:20:05] "GET / HTTP/1.1" 200 -
werkzeug - INFO - 127.0.0.1 - - [04/May/2022 23:20:05] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
werkzeug - INFO - 127.0.0.1 - - [04/May/2022 23:20:06] "GET / HTTP/1.1" 200 -
werkzeug - INFO - 127.0.0.1 - - [04/May/2022 23:20:06] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
werkzeug - INFO - 127.0.0.1 - - [04/May/2022 23:20:06] "GET / HTTP/1.1" 200 -
werkzeug - INFO - 127.0.0.1 - - [04/May/2022 23:20:06] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
werkzeug - INFO - 127.0.0.1 - - [04/May/2022 23:20:08] "GET /search?s=кот HTTP/1.1" 200 -
werkzeug - INFO - 127.0.0.1 - - [04/May/2022 23:20:08] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:21:02] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:21:02] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:21:05] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:21:05] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
ERROR:main:Exception on /posts [POST]
Traceback (most recent call last):
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:/Users/Admin/PycharmProjects/pythonProject2/main.py", line 51, in created_new_post
    return render_template("post_uploadet.html")
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\templating.py", line 149, in render_template
    ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 1081, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 1010, in get_template
    return self._load_template(name, globals)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 969, in _load_template
    template = self.loader.load(self, name, self.make_globals(globals))
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\loaders.py", line 126, in load
    source, filename, uptodate = self.get_source(environment, name)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\templating.py", line 59, in get_source
    return self._get_source_fast(environment, template)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\templating.py", line 95, in _get_source_fast
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: post_uploadet.html
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:21:12] "[35m[1mPOST /posts HTTP/1.1[0m" 500 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:25:13] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:25:13] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:25:18] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:25:18] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:25:31] "GET /search?s=другому HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:25:31] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:25:37] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:25:37] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
ERROR:main:Exception on /posts [POST]
Traceback (most recent call last):
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:/Users/Admin/PycharmProjects/pythonProject2/main.py", line 52, in created_new_post
    return render_template("post_uploadet.html")
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\templating.py", line 149, in render_template
    ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 1081, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 1010, in get_template
    return self._load_template(name, globals)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 969, in _load_template
    template = self.loader.load(self, name, self.make_globals(globals))
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\loaders.py", line 126, in load
    source, filename, uptodate = self.get_source(environment, name)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\templating.py", line 59, in get_source
    return self._get_source_fast(environment, template)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\templating.py", line 95, in _get_source_fast
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: post_uploadet.html
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:25:58] "[35m[1mPOST /posts HTTP/1.1[0m" 500 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:27:27] "POST /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:27:27] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:32:04] "GET /posts HTTP/1.1" 200 -
INFO:root:page add post
ERROR:main:Exception on /posts [POST]
Traceback (most recent call last):
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:/Users/Admin/PycharmProjects/pythonProject2/main.py", line 52, in created_new_post
    return render_template("post_uploaded.html")
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\templating.py", line 148, in render_template
    return _render(
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\templating.py", line 128, in _render
    rv = template.render(context)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 1301, in render
    self.environment.handle_exception()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\templates\post_uploaded.html", line 16, in top-level template code
    <img src="{{post.pic}}" width="100%" alt="">
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 485, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'post' is undefined
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:32:07] "[35m[1mPOST /posts HTTP/1.1[0m" 500 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:34:06] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:34:06] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:34:06] "[33mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 404 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:38:05] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:38:05] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:38:11] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:38:11] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:page add post
ERROR:main:Exception on /posts [POST]
Traceback (most recent call last):
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:/Users/Admin/PycharmProjects/pythonProject2/main.py", line 52, in created_new_post
    return render_template("post_uploaded.html")
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\templating.py", line 148, in render_template
    return _render(
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\templating.py", line 128, in _render
    rv = template.render(context)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 1301, in render
    self.environment.handle_exception()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\templates\post_uploaded.html", line 16, in top-level template code
    <img src="{{post.pic}}" width="100%" alt="">
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 485, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'post' is undefined
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:38:58] "[35m[1mPOST /posts HTTP/1.1[0m" 500 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:39:07] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:39:07] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:39:07] "[33mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 404 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:44:29] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:44:29] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:44:30] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:44:30] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
ERROR:main:Exception on /posts [POST]
Traceback (most recent call last):
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:/Users/Admin/PycharmProjects/pythonProject2/main.py", line 48, in created_new_post
    picture.save(picture_path)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\werkzeug\datastructures.py", line 2999, in save
    dst = open(dst, "wb")
PermissionError: [Errno 13] Permission denied: 'uploads/images/'
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:44:51] "[35m[1mPOST /posts HTTP/1.1[0m" 500 -
INFO:root:page add post
ERROR:main:Exception on /posts [POST]
Traceback (most recent call last):
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:/Users/Admin/PycharmProjects/pythonProject2/main.py", line 48, in created_new_post
    picture.save(picture_path)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\werkzeug\datastructures.py", line 2999, in save
    dst = open(dst, "wb")
PermissionError: [Errno 13] Permission denied: 'uploads/images/'
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:46:05] "[35m[1mPOST /posts HTTP/1.1[0m" 500 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:page add post
ERROR:main:Exception on /posts [POST]
Traceback (most recent call last):
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:/Users/Admin/PycharmProjects/pythonProject2/main.py", line 52, in created_new_post
    return render_template("post_uploaded.html")
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\templating.py", line 148, in render_template
    return _render(
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\templating.py", line 128, in _render
    rv = template.render(context)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 1301, in render
    self.environment.handle_exception()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\templates\post_uploaded.html", line 16, in top-level template code
    <img src="{{post.pic}}" width="100%" alt="">
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 485, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'post' is undefined
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:48:33] "[35m[1mPOST /posts HTTP/1.1[0m" 500 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:page add post
ERROR:main:Exception on /posts [POST]
Traceback (most recent call last):
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:/Users/Admin/PycharmProjects/pythonProject2/main.py", line 52, in created_new_post
    return render_template("post_uploaded.html")
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\templating.py", line 148, in render_template
    return _render(
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\templating.py", line 128, in _render
    rv = template.render(context)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 1301, in render
    self.environment.handle_exception()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\templates\post_uploaded.html", line 16, in top-level template code
    <img src="{{post.pic}}" width="100%" alt="">
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\jinja2\environment.py", line 485, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'post' is undefined
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:50:23] "[35m[1mPOST /posts HTTP/1.1[0m" 500 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:51:09] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:51:09] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:51:09] "[33mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 404 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:51:13] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:51:13] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:51:13] "[33mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 404 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:51:14] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:51:14] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [04/May/2022 23:51:14] "[33mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 404 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:10:23] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:10:24] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:10:24] "[33mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 404 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:10:28] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:10:28] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:10:50] "POST /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:10:50] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:10:50] "[33mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 404 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:11:07] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:11:07] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:16:36] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:16:36] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:16:36] "[33mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 404 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:16:45] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:16:45] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:17:02] "POST /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:17:02] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:17:02] "[33mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 404 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:18:02] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:18:34] "POST /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:18:34] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:18:34] "[33mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 404 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:22:02] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:22:02] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:22:31] "POST /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:22:31] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:22:31] "[33mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 404 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:25:31] "GET /posts HTTP/1.1" 200 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:25:37] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:25:37] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:25:37] "[33mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:25:37] "[33mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 404 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:25:42] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:25:42] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:26:26] "POST /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:26:26] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:26:26] "[33mGET /uploads/images/way-lake-mountains-1.jpg HTTP/1.1[0m" 404 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:27:03] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:27:03] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:27:06] "GET /search?s=Попытка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:27:06] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:27:06] "[33mGET /uploads/images/way-lake-mountains-1.jpg HTTP/1.1[0m" 404 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:27:12] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:27:12] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:27:23] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:27:23] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:29:10] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:29:10] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:29:13] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:29:13] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:29:19] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:29:19] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:29:51] "POST /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:29:51] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:29:51] "[33mGET /uploads/images/46344.jpg HTTP/1.1[0m" 404 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:29:56] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:29:56] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:31:45] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:31:45] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
ERROR:main:Exception on /posts [POST]
Traceback (most recent call last):
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:/Users/Admin/PycharmProjects/pythonProject2/main.py", line 47, in created_new_post
    picture.save(picture_path)
  File "C:\Users\Admin\PycharmProjects\pythonProject2\venv\lib\site-packages\werkzeug\datastructures.py", line 2999, in save
    dst = open(dst, "wb")
FileNotFoundError: [Errno 2] No such file or directory: '/uploads/images/455422.jpg'
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:32:22] "[35m[1mPOST /posts HTTP/1.1[0m" 500 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:35:59] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:35:59] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:36:00] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:36:00] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:36:00] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:36:00] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:36:01] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:36:01] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:36:01] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:36:01] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:36:01] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:36:01] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:36:01] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:36:01] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:37:09] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:37:09] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:37:12] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:37:12] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:37:12] "[33mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:37:12] "[33mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 404 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:37:21] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:37:22] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:root:No data uploaded
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:37:31] "POST /posts HTTP/1.1" 200 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:37:35] "GET /posts HTTP/1.1" 200 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:37:44] "POST /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:37:44] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [05/May/2022 00:37:44] "[33mGET /uploads/images/46344.jpg HTTP/1.1[0m" 404 -
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:werkzeug: * Running on http://127.0.0.1:800 (Press CTRL+C to quit)
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:18:38] "[33mGET /favicon.ico HTTP/1.1[0m" 404 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:18:41] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:18:41] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:18:44] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:18:44] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:18:47] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:18:47] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:19:00] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:19:00] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
ERROR:main:Exception on /uploads/images/hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads"], path)
KeyError: 'uploads'
ERROR:main:Exception on /uploads/images//hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads"], path)
KeyError: 'uploads'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:19:00] "[35m[1mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 500 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:19:00] "[35m[1mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 500 -
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:21:50] "GET /search?s=Картинка HTTP/1.1" 200 -
ERROR:main:Exception on /uploads/images/hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads"], path)
KeyError: 'uploads'
ERROR:main:Exception on /uploads/images//hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads"], path)
KeyError: 'uploads'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:21:50] "[35m[1mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 500 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:21:50] "[35m[1mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 500 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:21:50] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:21:50] "[33mGET /favicon.ico HTTP/1.1[0m" 404 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:22:02] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:22:02] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:22:03] "GET / HTTP/1.1" 200 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:22:08] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:22:08] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:22:12] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:22:12] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:22:17] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:22:17] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
ERROR:main:Exception on /uploads/images/hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads"], path)
KeyError: 'uploads'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:22:17] "[35m[1mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 500 -
ERROR:main:Exception on /uploads/images//hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads"], path)
KeyError: 'uploads'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:22:17] "[35m[1mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 500 -
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:23:12] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:23:12] "[33mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:23:12] "[33mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:23:12] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:23:13] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:23:13] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:23:13] "[33mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 22:23:13] "[33mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 404 -
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:03] "GET /search?s=Картинка HTTP/1.1" 200 -
ERROR:main:Exception on /uploads/images/hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads"], path)
KeyError: 'uploads'
ERROR:main:Exception on /uploads/images//hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads"], path)
KeyError: 'uploads'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:03] "[35m[1mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 500 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:03] "[35m[1mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 500 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:03] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:05] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:05] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
ERROR:main:Exception on /uploads/images/hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads"], path)
KeyError: 'uploads'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:05] "[35m[1mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 500 -
ERROR:main:Exception on /uploads/images//hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads"], path)
KeyError: 'uploads'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:05] "[35m[1mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 500 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:06] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:06] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
ERROR:main:Exception on /uploads/images/hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads"], path)
KeyError: 'uploads'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:06] "[35m[1mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 500 -
ERROR:main:Exception on /uploads/images//hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads"], path)
KeyError: 'uploads'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:06] "[35m[1mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 500 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:06] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:06] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
ERROR:main:Exception on /uploads/images/hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads"], path)
KeyError: 'uploads'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:06] "[35m[1mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 500 -
ERROR:main:Exception on /uploads/images//hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads"], path)
KeyError: 'uploads'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:06] "[35m[1mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 500 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:06] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:06] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
ERROR:main:Exception on /uploads/images/hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads"], path)
KeyError: 'uploads'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:06] "[35m[1mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 500 -
ERROR:main:Exception on /uploads/images//hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads"], path)
KeyError: 'uploads'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:06] "[35m[1mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 500 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:07] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:07] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
ERROR:main:Exception on /uploads/images/hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads"], path)
KeyError: 'uploads'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:07] "[35m[1mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 500 -
ERROR:main:Exception on /uploads/images//hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads"], path)
KeyError: 'uploads'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:07] "[35m[1mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 500 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:10] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:10] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:12] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:12] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:17] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:17] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:19] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:19] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
ERROR:main:Exception on /uploads/images/hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads"], path)
KeyError: 'uploads'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:19] "[35m[1mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 500 -
ERROR:main:Exception on /uploads/images//hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads"], path)
KeyError: 'uploads'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:13:19] "[35m[1mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 500 -
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:14:00] "GET /search?s=Картинка HTTP/1.1" 200 -
ERROR:main:Exception on /uploads/images/hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads/image/"], path)
KeyError: 'uploads/image/'
ERROR:main:Exception on /uploads/images//hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads/image/"], path)
KeyError: 'uploads/image/'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:14:00] "[35m[1mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 500 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:14:00] "[35m[1mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 500 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:14:00] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:14:01] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:14:01] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
ERROR:main:Exception on /uploads/images/hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads/image/"], path)
KeyError: 'uploads/image/'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:14:01] "[35m[1mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 500 -
ERROR:main:Exception on /uploads/images//hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads/image/"], path)
KeyError: 'uploads/image/'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:14:01] "[35m[1mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 500 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:14:03] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:14:03] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
ERROR:main:Exception on /uploads/images/hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads/image/"], path)
KeyError: 'uploads/image/'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:14:03] "[35m[1mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 500 -
ERROR:main:Exception on /uploads/images//hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads/image/"], path)
KeyError: 'uploads/image/'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:14:03] "[35m[1mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 500 -
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:15:21] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:15:21] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:15:42] "POST /posts HTTP/1.1" 200 -
ERROR:werkzeug:127.0.0.1 - - [08/May/2022 23:15:42] code 400, message Bad request syntax ('------WebKitFormBoundarytN3KNDkmiMLhYtZq')
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:15:42] "[35m[1mNone /posts HTTP/0.9[0m" HTTPStatus.BAD_REQUEST -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:15:42] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:18:30] "POST /posts HTTP/1.1" 200 -
ERROR:werkzeug:127.0.0.1 - - [08/May/2022 23:18:30] code 400, message Bad request syntax ('------WebKitFormBoundaryIubrVft4hpfub7Z4')
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:18:30] "[35m[1mNone /posts HTTP/0.9[0m" HTTPStatus.BAD_REQUEST -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:18:30] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:31:25] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:31:25] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:31:27] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:31:27] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:31:32] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:31:32] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:31:50] "POST /posts HTTP/1.1" 200 -
ERROR:werkzeug:127.0.0.1 - - [08/May/2022 23:31:50] code 400, message Bad request syntax ('------WebKitFormBoundaryR7vWIWDvJbNRZMlT')
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:31:50] "[35m[1mNone /posts HTTP/0.9[0m" HTTPStatus.BAD_REQUEST -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:31:50] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:32:34] "POST /posts HTTP/1.1" 200 -
ERROR:werkzeug:127.0.0.1 - - [08/May/2022 23:32:34] code 400, message Bad request syntax ('------WebKitFormBoundarymg76CzjHFYHFQpNq')
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:32:34] "[35m[1mNone /posts HTTP/0.9[0m" HTTPStatus.BAD_REQUEST -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:32:34] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:34:38] "POST /posts HTTP/1.1" 200 -
ERROR:werkzeug:127.0.0.1 - - [08/May/2022 23:34:38] code 400, message Bad request syntax ('------WebKitFormBoundarymg76CzjHFYHFQpNq')
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:34:38] "[35m[1mNone /posts HTTP/0.9[0m" HTTPStatus.BAD_REQUEST -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:34:38] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:35:03] "POST /posts HTTP/1.1" 200 -
ERROR:werkzeug:127.0.0.1 - - [08/May/2022 23:35:03] code 400, message Bad request syntax ('------WebKitFormBoundaryAPGRBsX5B9lfYqeC')
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:35:03] "[35m[1mNone /posts HTTP/0.9[0m" HTTPStatus.BAD_REQUEST -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:35:03] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:37:55] "POST /posts HTTP/1.1" 200 -
ERROR:werkzeug:127.0.0.1 - - [08/May/2022 23:37:55] code 400, message Bad request syntax ('------WebKitFormBoundarydMh3yAeMFpbjDDOE')
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:37:55] "[35m[1mNone /posts HTTP/0.9[0m" HTTPStatus.BAD_REQUEST -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:37:55] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:38:49] "POST /posts HTTP/1.1" 200 -
ERROR:werkzeug:127.0.0.1 - - [08/May/2022 23:38:49] code 400, message Bad request syntax ('------WebKitFormBoundaryBHu0X73AWAJDP5yC')
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:38:49] "[35m[1mNone /posts HTTP/0.9[0m" HTTPStatus.BAD_REQUEST -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:38:49] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:38:51] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:38:51] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:38:54] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:38:54] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
ERROR:main:Exception on /uploads/images/hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["/uploads/"], path)
KeyError: '/uploads/'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:38:54] "[35m[1mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 500 -
ERROR:main:Exception on /uploads/images//hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["/uploads/"], path)
KeyError: '/uploads/'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:38:54] "[35m[1mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 500 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:38:56] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:38:56] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:39:16] "POST /posts HTTP/1.1" 200 -
ERROR:werkzeug:127.0.0.1 - - [08/May/2022 23:39:16] code 400, message Bad request syntax ('------WebKitFormBoundaryHGmXPQskk1Fn8sQE')
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:39:16] "[35m[1mNone /posts HTTP/0.9[0m" HTTPStatus.BAD_REQUEST -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:39:16] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:40:25] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:40:25] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:40:35] "GET /search?s=кот HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:40:35] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:40:37] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:40:37] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:40:39] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:40:39] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
ERROR:main:Exception on /posts [POST]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 51, in created_new_post
    return render_template("post_uploaded.html", new_post=new_post)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\templating.py", line 148, in render_template
    return _render(
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\templating.py", line 128, in _render
    rv = template.render(context)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\jinja2\environment.py", line 1291, in render
    self.environment.handle_exception()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\jinja2\environment.py", line 926, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\templates\post_uploaded.html", line 16, in top-level template code
    <img src="{{ url_for(new_post.pic) }}" width="100%" alt="">
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\helpers.py", line 336, in url_for
    return appctx.app.handle_url_build_error(error, endpoint, values)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\helpers.py", line 323, in url_for
    rv = url_adapter.build(
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\werkzeug\routing.py", line 2307, in build
    raise BuildError(endpoint, values, method, self)
werkzeug.routing.BuildError: Could not build url for endpoint 'uploads/images/455422.jpg'. Did you mean 'main_blueprint.main_page' instead?
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:40:56] "[35m[1mPOST /posts HTTP/1.1[0m" 500 -
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:root:page add post
ERROR:main:Exception on /posts [POST]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 51, in created_new_post
    return render_template("post_uploaded.html", new_post=new_post)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\templating.py", line 148, in render_template
    return _render(
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\templating.py", line 128, in _render
    rv = template.render(context)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\jinja2\environment.py", line 1291, in render
    self.environment.handle_exception()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\jinja2\environment.py", line 926, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\templates\post_uploaded.html", line 16, in top-level template code
    <img src="{{ url_for(new_post.pic) }}" width="100%" alt="">
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\helpers.py", line 336, in url_for
    return appctx.app.handle_url_build_error(error, endpoint, values)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\helpers.py", line 323, in url_for
    rv = url_adapter.build(
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\werkzeug\routing.py", line 2307, in build
    raise BuildError(endpoint, values, method, self)
werkzeug.routing.BuildError: Could not build url for endpoint 'uploads/images/455422.jpg'. Did you mean 'main_blueprint.main_page' instead?
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:43:26] "[35m[1mPOST /posts HTTP/1.1[0m" 500 -
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:44:25] "POST /posts HTTP/1.1" 200 -
ERROR:main:Exception on /uploads/images/455422.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads/"], path)
KeyError: 'uploads/'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:44:25] "[35m[1mGET /uploads/images/455422.jpg HTTP/1.1[0m" 500 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:44:25] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:45:37] "GET /posts HTTP/1.1" 200 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:45:39] "POST /posts HTTP/1.1" 200 -
ERROR:main:Exception on /uploads/images/455422.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads/images/"], path)
KeyError: 'uploads/images/'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:45:39] "[35m[1mGET /uploads/images/455422.jpg HTTP/1.1[0m" 500 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:45:39] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:45:46] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:45:46] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:45:49] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:45:49] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
ERROR:main:Exception on /uploads/images/hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads/images/"], path)
KeyError: 'uploads/images/'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:45:49] "[35m[1mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 500 -
ERROR:main:Exception on /uploads/images//hdr009.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads/images/"], path)
KeyError: 'uploads/images/'
ERROR:main:Exception on /uploads/images/455422.jpg [GET]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 56, in static_dir
    return send_from_directory(app.config["uploads/images/"], path)
KeyError: 'uploads/images/'
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:45:49] "[35m[1mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 500 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:45:49] "[35m[1mGET /uploads/images/455422.jpg HTTP/1.1[0m" 500 -
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:46:53] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:46:53] "[33mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:46:53] "[33mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:46:53] "[33mGET /uploads/images/455422.jpg HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:46:53] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:46:57] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:46:57] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:46:57] "[33mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:46:57] "[33mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:46:57] "[33mGET /uploads/images/455422.jpg HTTP/1.1[0m" 404 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:47:00] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:47:00] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:47:15] "POST /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:47:15] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:47:15] "[33mGET /uploads/images/way-lake-mountains-1.jpg HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:57:20] "[33mGET /uploads/images/way-lake-mountains-1.jpg HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:57:34] "[33mGET /uploads/images/way-lake-mountains-1.jpg HTTP/1.1[0m" 404 -
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:59:07] "POST /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:59:07] "[33mGET /uploads/images/way-lake-mountains-1.jpg HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:59:07] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:59:08] "POST /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:59:08] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:59:08] "[33mGET /uploads/images/way-lake-mountains-1.jpg HTTP/1.1[0m" 404 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:59:09] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:59:09] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:59:11] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:59:11] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:59:12] "[33mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:59:12] "[33mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:59:12] "[33mGET /uploads/images/455422.jpg HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:59:12] "[33mGET /uploads/images/way-lake-mountains-1.jpg HTTP/1.1[0m" 404 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:59:15] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:59:15] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:59:36] "POST /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:59:36] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [08/May/2022 23:59:36] "[33mGET /uploads/images/way-lake-mountains-1.jpg HTTP/1.1[0m" 404 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [09/May/2022 15:39:46] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/May/2022 15:39:46] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [09/May/2022 18:17:41] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/May/2022 18:17:41] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [09/May/2022 20:04:06] "GET /posts HTTP/1.1" 200 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [09/May/2022 20:04:08] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/May/2022 20:04:08] "GET /uploads/images//hdr009.jpg HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/May/2022 20:04:08] "GET /uploads/images/hdr009.jpg HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/May/2022 20:04:08] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [09/May/2022 20:04:08] "GET /uploads/images/455422.jpg HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/May/2022 20:04:08] "GET /uploads/images/way-lake-mountains-1.jpg HTTP/1.1" 200 -
INFO:root:A search is being performed
INFO:werkzeug:127.0.0.1 - - [09/May/2022 20:04:20] "GET /search?s=Картинка HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/May/2022 20:04:20] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [09/May/2022 20:04:20] "[36mGET /uploads/images/hdr009.jpg HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [09/May/2022 20:04:20] "[36mGET /uploads/images//hdr009.jpg HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [09/May/2022 20:04:20] "[36mGET /uploads/images/455422.jpg HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [09/May/2022 20:04:20] "[36mGET /uploads/images/way-lake-mountains-1.jpg HTTP/1.1[0m" 304 -
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:root:Loading the main page
INFO:werkzeug:127.0.0.1 - - [09/May/2022 20:41:05] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/May/2022 20:41:06] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [09/May/2022 20:41:09] "GET /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/May/2022 20:41:09] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:root:page add post
ERROR:main:Exception on /posts [POST]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 50, in created_new_post
    functions.overwrite_json_files(posts, data_file)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\functions.py", line 10, in overwrite_json_files
    with open(path, "w", encoding="utf-8", ensure_ascii=False) as file:
TypeError: 'ensure_ascii' is an invalid keyword argument for open()
INFO:werkzeug:127.0.0.1 - - [09/May/2022 20:41:39] "[35m[1mPOST /posts HTTP/1.1[0m" 500 -
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:root:page add post
ERROR:main:Exception on /posts [POST]
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\main.py", line 50, in created_new_post
    functions.overwrite_json_files(posts, data_file)
  File "C:\Users\Admin\PycharmProjects\Domashka12.1\Flask_Domashka12\functions.py", line 10, in overwrite_json_files
    with open(path, "w",  ensure_ascii=False) as file:
TypeError: 'ensure_ascii' is an invalid keyword argument for open()
INFO:werkzeug:127.0.0.1 - - [09/May/2022 20:42:22] "[35m[1mPOST /posts HTTP/1.1[0m" 500 -
INFO:werkzeug: * Running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:root:page add post
INFO:werkzeug:127.0.0.1 - - [09/May/2022 20:43:04] "POST /posts HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/May/2022 20:43:04] "[36mGET /static/style.css HTTP/1.1[0m" 304 -
INFO:werkzeug:127.0.0.1 - - [09/May/2022 20:43:04] "GET /uploads/images/46344.jpg HTTP/1.1" 200 -
